dagshub/colab/login.py

```diff
@@ -0,0 +1,33 @@
+from dagshub.auth import add_oauth_token, get_token
+from dagshub.auth.tokens import get_user_of_token
+from dagshub.common.api import RepoAPI
+from dagshub.common.api.repo import RepoNotFoundError
+from dagshub.common.helpers import log_message
+from dagshub.upload import create_repo
+
+COLAB_REPO_NAME = "dagshub-drive"
+
+
+def login() -> str:
+    """
+    Run custom colab-specific flow, which helps with setting up a repository,
+    storage of which will be used as an alternative to Google Drive.
+
+    Returns the name of the repository that can be used with colab-related functionality (``<user>/dagshub-drive``)
+    """
+    try:
+        token = get_token(fail_if_no_token=True)
+    except RuntimeError:
+        add_oauth_token(referrer="colab")
+        token = get_token()
+
+    username = get_user_of_token(token)
+
+    colab_repo = RepoAPI(f"{username}/{COLAB_REPO_NAME}")
+    try:
+        colab_repo.get_repo_info()
+    except RepoNotFoundError:
+        create_repo(COLAB_REPO_NAME)
+    log_message(f"Repository {colab_repo.full_name} is ready for use with Colab. Link to the repository:")
+    print(colab_repo.repo_url)
```

Comment:
Why is one of them `log_message` and the other `print`?

Reply:
The colab/ipython printer works inconsistently with the URLs and not always makes them clickable, so whenever a URL is printed that we want users to be able to click AND we know that it will probably show up in colab, we use regular print.

Other examples:
https://github.com/DagsHub/client/blob/0c2d1ff58037fc0b897d3f9e8d6609ded4c477c9/dagshub/auth/oauth.py#L51
https://github.com/DagsHub/client/blob/bf3832a0350ab9400ee50ad2b25e1da9ba2f8da6/dagshub/data_engine/model/datasource.py#L841

Honestly I can probably replace the log_message with a print for consistency sake.

Reply:
log_message prints to rich's console which does neat highlights for numbers/urls on some occasions, so I thought to use it here for neater output